Library/Homebrew/keg_relocate.rb

```diff
@@ -87,16 +87,25 @@ def change_rpath(file, new_prefix)
     end
     return unless patchelf.installed?
     cmd = "#{patchelf.bin}/patchelf --set-rpath #{new_prefix}/lib"
-    (old_i, old_p) = `#{patchelf.bin}/patchelf --print-rpath --print-interpreter #{file}`.split("\n")
+    old_rpath = `#{patchelf.bin}/patchelf --print-rpath #{file}`.strip
     raise ErrorDuringExecution.new(cmd) unless $?.success?
+    lib_path = "#{new_prefix}/lib"
+    rpath = old_rpath.split(":").select do |x|
+      x.start_with?(old_prefix) || x.start_with?(new_prefix)
+    end.map { |x| x.sub(old_prefix, new_prefix) }
```

Comment:
This could be made a bit more terse if you swap the `select` and the `map`:
```ruby
rpath = old_rpath.split(":").map { |x| x.sub(old_prefix, new_prefix) }.select { |x| x.start_with?(new_prefix) }
```